### PR TITLE
Fixing test suite

### PR DESCRIPTION
**Pull request by eturino**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -23,12 +27,19 @@ group :development do
   gem 'rake'
   # gem 'debugger'
   gem 'quiet_assets'
-  gem 'artirix_gem_release', :git => "git@gitlab.artirix.com:artirix/artirix_gem_release.git"
+
+  # ignoring artirix_gem_release in public github repo
+  # gem 'artirix_gem_release', :git => "git@gitlab.artirix.com:artirix/artirix_gem_release.git"
```
Comment (reuben-sutton) [inline Gemfile]: Does it make sense to release a public github repo into our private gem server?

File: Rakefile [hunk 2]
```diff
@@ -11,7 +11,10 @@ rescue LoadError
   require 'rake/rdoctask'
   RDoc::Task = Rake::RDocTask
 end
-require 'artirix_gem_release'
+
+# ignoring artirix_gem_release in public github repo
+# require 'artirix_gem_release'
```
Comment (reuben-sutton) [inline Rakefile]: Same as above

File: app/inputs/attachments_input.rb [hunk 3]
```diff
@@ -1,6 +1,6 @@
 class AttachmentsInput < SimpleForm::Inputs::Base
 
-  def input(wrapper_options)
+  def input(_wrapper_options = nil)
```
Comment (reuben-sutton) [inline app/inputs/attachments_input.rb:3]: Why add the underscore?

File: app/models/cms/view_context.rb [hunk 4]
```diff
@@ -46,9 +46,7 @@ def method_missing(method, *args, &block)
 
       # try mounted helpers (note: we prefer mounted helper than routes.url_helpers)
       mounted_helper_method = "_#{method}"
-      if @_mounted_helpers && @_mounted_helpers.respond_to?(mounted_helper_method)
-        return send(mounted_helper_method, *args, &block)
-      end
+      return send(mounted_helper_method, *args, &block) if @_mounted_helpers && respond_to?(mounted_helper_method)
```
Comment (reuben-sutton) [inline app/models/cms/view_context.rb]: I wonder if the length of this line diminishes readability?

File: features/step_definitions/deprecated.rb [hunk 5]
```diff
@@ -1,7 +1,3 @@
-Given /^we are using a Rails 4.0 compatible version of cucumber$/ do
-  pending
-end
-
 Given /^I'm creating content which uses deprecated input fields$/ do
```
Comment (reuben-sutton) [inline features/step_definitions/deprecated.rb:1]: Would it be reasonable to rid our version of deprecated features?

File: features/support/env.rb [hunk 6]
```diff
@@ -1,5 +1,9 @@
 # Added for BrowserCMS (to test using test/dummy app)
 ENV["RAILS_ENV"] ||= "test"
+
+require "codeclimate-test-reporter"
+CodeClimate::TestReporter.start
```
Comment (reuben-sutton) [inline features/support/env.rb:5]: :+1:

File: features/step_definitions/deprecated.rb (hunk 5 above)
Comment (eturino) [inline features/step_definitions/deprecated.rb:1]: This was only for making it green. I'm working in another branch for killing half of bcms (everything that we don't use)

File: features/support/env.rb (hunk 6 above)
Comment (eturino) [inline features/support/env.rb:5]: the test coverage does not work :( 

CodeClimate can only receive 1 input per commit, and this generates 2 (one for test, another for cucumber).

In any case, it's quite bad, ~80% or so.

File: app/inputs/attachments_input.rb (hunk 3 above)
Comment (eturino) [inline app/inputs/attachments_input.rb:3]: convention for "unused variable" is to call it `_` or `_whatever`

**Pull request merged**